assets/src/hooks/useNotifications.ts

```diff
@@ -1,12 +1,21 @@
 import { Socket } from "phoenix"
+import { array, Infer, object, union } from "superstruct"
 import {
   NotificationData,
   notificationFromData,
 } from "../models/notificationData"
 import { Notification } from "../realtime.d"
-import { useChannel } from "./useChannel"
+import { useCheckedChannel } from "./useChannel"
 
-type InitialNotificationData = { initial_notifications: NotificationData[] }
+const InitialNotificationData = object({
+  initial_notifications: array(NotificationData),
```

Comment:
The tricky thing is that here we're using `NotificationData` as a value, not a type, so `NotificationData[]` wouldn't even be syntactically correct. My understanding is that these various Superstruct functions like `object` and `array` are composed to produce an object that contains the methods to validate given data, but if you're using Typescript it's also smart about the resulting types.

Reply:
Thanks! I was interested from thinking about if I go to go to add a new `object` somewhere for the first time - I'm sure it will take some muscle memory for me to remember to use `array` but glad to know the typing will prevent those kinds of mistakes!